pubsub.go

```diff
@@ -87,27 +121,37 @@ func (p *PubsubValueStore) PutValue(ctx context.Context, key string, value []byt
 	}
 
 	log.Debugf("PubsubPublish: publish value for key", key)
-	return p.ps.Publish(topic, value)
+	done := make(chan error, 1)
+	go func() {
+		done <- p.ps.Publish(topic, value)
+	}()
+
+	select {
+	case err := <-done:
+		return err
+	case <-ctx.Done():
+		return ctx.Err()
+	}
 }
 
-func (p *PubsubValueStore) isBetter(key string, val []byte) bool {
+func (p *PubsubValueStore) isBetter(key string, val []byte) (isBetter bool, isEqual bool) {
 	if p.Validator.Validate(key, val) != nil {
-		return false
+		return false, false
 	}
 
 	old, err := p.getLocal(key)
 	if err != nil {
 		// If the old one is invalid, the new one is *always* better.
-		return true
+		return true, false
 	}
 
-	// Same record. Possible DoS vector, should consider failing?
```

Comment:
I agree, this seems bizarre. Let's at least return that they are equal instead of deciding equal = better.